test/unittest/unittest_nntrainer_models.cpp

```diff
@@ -423,18 +429,23 @@ void GraphWatcher::compareFor(const std::string &reference,
                   nntrainer::Tensor::epsilon);
     }
 
-    for (auto it = nodes.begin(); it != nodes.end() - 1; ++it) {
-      it->forward(iteration, *(it + 1));
+    auto it = nodes.begin();
+    for (; it != nodes.end() - 1; ++it) {
+      it->forward(iteration, !(it + 1)->supportInPlace());
     }
+    it->forward(iteration, true);
 
     if (loss_nodes.size()) {
       nn.backwarding(label, iteration);
 
-      for (auto it = nodes.rbegin(); it != nodes.rend() - 1; it++) {
-        if (it == nodes.rend())
-          it->backward(iteration, true, !optimize);
-        else
+      for (auto it = nodes.rbegin(); it != nodes.rend(); it++) {
+        if (it == nodes.rend() - 1) {
+          /** check last layer backwarding only when not input layers */
+          if (it->getNodeType() != nntrainer::InputLayer::type)
```

Comment:
Could you change this to query `getNumInputConnections()`?

Reply:
This is intentional. First layer can be convolution but we want to verify backwarding for it.
Updated rather to supportBackwarding.

Reply:
I see :)